# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../wvstreams/glibc212.patch
# Copyright (C) 2011 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

http://bugs.gentoo.org/333301

--- wvstreams-4.6.1/ipstreams/wvunixdgsocket.cc
+++ wvstreams-4.6.1/ipstreams/wvunixdgsocket.cc
@@ -1,5 +1,5 @@
 #include "wvunixdgsocket.h"
-#ifdef MACOS
+#if defined(MACOS) || defined(__GNUC__)
 #include <sys/types.h>
 #include <sys/stat.h>
 #endif
--- wvstreams-4.6.1/streams/wvatomicfile.cc
+++ wvstreams-4.6.1/streams/wvatomicfile.cc
@@ -11,7 +11,8 @@
 #include "wvfileutils.h"
 #include "wvstrutils.h"
 
-#ifdef MACOS
+#if defined(MACOS) || defined(__GNUC__)
+#include <sys/types.h>
 #include <sys/stat.h>
 #endif
 

